Cabal/Distribution/Simple/Setup.hs

```diff
@@ -1942,6 +1944,11 @@ testOptions' showOrParseArgs =
         "keep .tix files for HPC between test runs"
         testKeepTix (\v flags -> flags { testKeepTix = v})
         trueArg
+  , option [] ["test-wrapper"]
+        "Run test through a wrapper."
+        testWrapper (\v flags -> flags { testWrapper = v })
+        (reqArg' "FILE" (toFlag :: FilePath -> Flag FilePath)
+            (flagToList :: Flag FilePath -> [FilePath]))
```

Comment:
You need to add this to at least `Distribution.Cabal.Setup.testOptions` if you want it to actually make it through to a user-facing option.